crates/subspace-core-primitives/src/segments.rs

```diff
@@ -71,16 +71,40 @@ impl SegmentIndex {
         PieceIndex::from(self.0 * ArchivedHistorySegment::NUM_PIECES as u64)
     }
 
+    /// Get the last piece index in this segment.
+    pub fn last_piece_index(&self) -> PieceIndex {
+        PieceIndex::from((self.0 + 1) * ArchivedHistorySegment::NUM_PIECES as u64 - 1)
+    }
+
     /// Iterator over piece indexes that belong to this segment.
```

Comment:
It's not an iterator any more.

Reply:
By the way, why not iterator?

Reply:
As commit message says, it is such that we can iterate over it in parallel later. Compiler should generate an efficient code for statically sized array, so we don't have to `.collect::<Vec<_>>()` it before being able to iterate in parallel. Essentially no extra heap allocations anymore.